### ISSUE TITLE
support for window.performance.navigation

### ISSUE THREAD

**Issue by dgoldstein0** [jsdom/jsdom#2136]
### Basic info:

- **Node.js version:** 8.9.3
- **jsdom version:** 11.6.2

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM('', {});
console.log(dom.window.performance.navigation.TYPE_NAVIGATE) // should print 0
```
### How does similar code behave in browsers?
https://developer.mozilla.org/en-US/docs/Web/API/Performance/navigation
https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation

`window.performance.navigation` is an object that always has 4 static members (TYPE_NAVIGATE, TYPE_RELOAD, TYPE_BACK_FORWARD, and TYPE_RESERVED) and two members - `type` and `redirectCount`.  I think these detail how the page was navigated to.  There's also a `toJSON` method worth mentioning.

I guess I'm not totally sure how type/redirectCount should behave in jsdom, but TYPE_NAVIGATE et. al. should always be present.

I'm mainly filing this issue (a) to track this and (b) so we can spec the desired behavior; my guess is it'll be easy to implement once we spec the desired behavior.

**Comment by domenic** [comment]
I found where these are all defined: https://w3c.github.io/navigation-timing/

It's a pretty large spec, and I'm not sure that implementing, piecemeal, four members marked obsolete is a good idea. Additionally, properly implementing the spec would require implementing https://w3c.github.io/performance-timeline/ and https://w3c.github.io/resource-timing/. That seems like a lot of work. I'd suggest instead just stubbing these out if you have tests or similar depending on these obsolete properties, using https://github.com/jsdom/jsdom#intervening-before-parsing

**Comment by dgoldstein0** [comment]
Fair enough.  I was hoping not to need to maintain a shim, but what I need
at the moment is small in comparison to the spec so I'll just go that route
for now.
